### PR TITLE
NEXUS-21956 Helm. Hosted repository does not work with tgz files crea…

### PR DESCRIPTION
**Pull request by mlukaretkyi**
https://issues.sonatype.org/browse/NEXUS-21956

### CODE REVIEW

File: nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/util/TgzParserTest.java [hunk 1]
```diff
@@ -12,39 +12,58 @@
...
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.core.Is.is;
-import static org.mockito.Mockito.when;
+import org.sonatype.goodies.testsupport.TestSupport;
 
 public class TgzParserTest
-    extends TestSupport
-{
-  private TgzParser underTest;
+    extends TestSupport {
```
Comment (sergeyonofreychuck) [inline nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/util/TgzParserTest.java]: "{" should be on the next line.

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/util/TgzParser.java [hunk 2]
```diff
@@ -33,32 +29,25 @@
...
     try (GzipCompressorInputStream gzis = new GzipCompressorInputStream(is)) {
       try (TarArchiveInputStream tais = new TarArchiveInputStream(gzis)) {
-        ArchiveEntry currentEntry = tais.getNextEntry();
-
-        while (currentEntry != null) {
-          if (currentEntry.getName().contains(CHART_NAME)) {
+        ArchiveEntry currentEntry;
+        while ((currentEntry = tais.getNextEntry()) != null) {
+          String entryName = currentEntry.getName();
+          if (entryName.contains(CHART_NAME) &&
```
Comment (sergeyonofreychuck) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/util/TgzParser.java]: entryName.endswith ?

Comment (sergeyonofreychuck) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/util/TgzParser.java]: only two possible names you need "Chart.yaml" and "./Chart.yaml"

Comment (mlukaretkyi) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/util/TgzParser.java]: Yes, you are right. endsWith will be better. 
https://github.com/sonatype-nexus-community/nexus-repository-helm/pull/99/commits/0b7d5ad18bc2fa72e47784a81af8829b4e887ee7

Possible, valid names
chart-name/Chart.yaml

mysql/Chart.yaml
mongodb/Chart.yaml

File: nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/util/TgzParserTest.java (hunk 1 above)
Comment (mlukaretkyi) [reply nexus-repository-helm/src/test/java/org/sonatype/repository/helm/internal/util/TgzParserTest.java]: fixed

**Pull request merged**